### DISCUSSION TITLE
Uploaded files are showing up as 0 bytes in Supabase Storage - React Native Expo

### DISCUSSION THREAD

**Discussion by Atharva2628** [supabase/supabase#2336] [Questions] [answered]
Hi,

I have been working on a project on Expo - React Native through Supabase which involves the upload of images to storage buckets. Although, when I upload files to supabase, they show up as the correct format, name, etc. but with the size of 0 bytes. I'm using ImagePicker from Expo to select images from the library.
Here's my code: 

### Image Selection:
```
    let result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: ImagePicker.MediaTypeOptions.All,
        allowsEditing: true,
        base64: true,
        quality: 0.5,
      });
```

### Conversion of base64 to File:
```
    var file = dataURLtoFile(
      "data:image/jpg;base64," + result.base64,
      "test.jpg"
    );
```
(The size of the converted file up until here is correct [791496 bytes] )

**Upload Statement:**
```
  let { data, error } = await supabase.storage
        .from("avatars")
        .upload(file.name, file);
      console.log(data, error);
```

(The file uploads with no errors, only it doesn't show up properly in the supabase browser panel [0 bytes] )

[
![test](https://user-images.githubusercontent.com/85646821/125303631-90b81400-e34a-11eb-87b9-d838ad12df6b.PNG)
](url)

**Comment by thebengeu** [comment, marked as answer]
I'm not sure how to duplicate this, could you share the contents of your `dataURLtoFile` function please?

**Reply by thebengeu** [reply]
I was able to duplicate this issue: when using `File`, the uploaded file is fine for web but 0 bytes for iOS.

We added support for `ArrayBuffer` and other types [yesterday](https://github.com/supabase/storage-js/issues/5#issuecomment-882583732). I've tested that the following works:

```js
import {decode} from 'base64-arraybuffer'

const result = await ImagePicker.launchImageLibraryAsync({base64: true})
const { data, error } = await supabase.storage
  .from('avatars')
  .upload('test.jpg', decode(result.base64), {contentType: 'image/jpg'})
```

**Reply by Atharva2628** [reply]
Thanks mate, worked like a charm 😀🎉
